src/context/wishlist-context.js

```diff
@@ -0,0 +1,21 @@
+import { createContext, useReducer, useContext } from "react";
+
+import { wishlistReducerFn } from "../reducer/wishlistReducerFn";
+
+const wishlistContext = createContext();
+
+const useWishlistContext = () => useContext(wishlistContext);
+
+const WishlistProvider = ({ children }) => {
+  const [wishlistState, wishlistDispatch] = useReducer(wishlistReducerFn, {
+    wishlistArray: [],
+  });
+
+  return (
+    <wishlistContext.Provider value={{ wishlistState, wishlistDispatch }}>
```

Comment:
Instead of passing the entire state you can simplify and pass only the state variable

```suggestion
    <wishlistContext.Provider value={{ wishlist: wishlistState.wishlistArray, wishlistDispatch }}>
```

And in the component you can do as below

```JS
const {wishlist} = useWishlistContext()
```

Reply:
This is outdated code, I am not using wishlist context at all, I have removed it now